### ISSUE TITLE
[explicit-function-return-type] Conflicts with typescript rule on setters

### ISSUE THREAD

**Issue by calling** [typescript-eslint/typescript-eslint#525]
**Repro**

```JSON
{
  "rules": {
    "typescript/explicit-function-return-type": ["error"]
  }
}
```

```TS
const myObj = {
  set myProp(val) {
    this.myProp = val;
  },
};

```

**Expected Result**
No error. Adding an explicit return type causes the following TS error: `A 'set' accessor cannot have a return type annotation.ts(1095)`
**Actual Result**
`Missing return type on function.eslint(@typescript-eslint/explicit-function-return-type)`
**Additional Info**

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `1.7.0` |
| `@typescript-eslint/parser`        | `1.7.0` |
| `TypeScript`                       | `3.4.5` |
| `ESLint`                           | `5.3.0` |
| `node`                             | `10.15.3` |
| `npm`                              | `6.9.0` |
